### ISSUE TITLE
Error Installing using Carthage

### ISSUE THREAD

**Issue by chamitha** [davidstump/SwiftPhoenixClient#164]
Carthage is throwing an error when installing;

```
*** Building scheme "SwiftPhoenixClient" in ChatExample.xcworkspace
Build Failed
	Task failed with exit code 65:
	/usr/bin/xcrun xcodebuild -workspace /Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Example/ChatExample.xcworkspace -scheme SwiftPhoenixClient -configuration Release -derivedDataPath /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0 -sdk iphoneos ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive -archivePath /var/folders/l4/l18my7x54g96szlt2y2q387m0000gn/T/SwiftPhoenixClient SKIP_INSTALL=YES GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO CLANG_ENABLE_CODE_COVERAGE=NO STRIP_INSTALLED_PRODUCT=NO (launched in /Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient)

This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/l4/l18my7x54g96szlt2y2q387m0000gn/T/carthage-xcodebuild.PQsqUL.log
```

I'm using Xcode 11.3 with Build Tools. The relevant excerpt from the log is;

```
CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler (in target 'SwiftPhoenixClient' from project 'Pods')
    cd /Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Example/Pods
    export DEVELOPER_DIR=/Applications/Xcode.app/Contents/Developer
    export SDKROOT=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.2.sdk
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc -incremental -module-name SwiftPhoenixClient -O -whole-module-optimization -enforce-exclusivity=checked @/Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/Objects-normal/arm64/SwiftPhoenixClient.SwiftFileList -D COCOAPODS -suppress-warnings -sdk /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.2.sdk -target arm64-apple-ios9.0 -g -module-cache-path /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/ModuleCache.noindex -Xfrontend -serialize-debugging-options -embed-bitcode -swift-version 5 -I /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/BuildProductsPath/Release-iphoneos/SwiftPhoenixClient -F /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/BuildProductsPath/Release-iphoneos/SwiftPhoenixClient -F /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/BuildProductsPath/Release-iphoneos/Starscream -c -num-threads 8 -output-file-map /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/Objects-normal/arm64/SwiftPhoenixClient-OutputFileMap.json -parseable-output -serialize-diagnostics -emit-dependencies -emit-module -emit-module-path /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/Objects-normal/arm64/SwiftPhoenixClient.swiftmodule -Xcc -I/Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/swift-overrides.hmap -Xcc -iquote -Xcc /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/SwiftPhoenixClient-generated-files.hmap -Xcc -I/Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/SwiftPhoenixClient-own-target-headers.hmap -Xcc -I/Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/SwiftPhoenixClient-all-non-framework-target-headers.hmap -Xcc -ivfsoverlay -Xcc /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/all-product-headers.yaml -Xcc -iquote -Xcc /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/SwiftPhoenixClient-project-headers.hmap -Xcc -I/Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/BuildProductsPath/Release-iphoneos/SwiftPhoenixClient/include -Xcc -I/Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/DerivedSources-normal/arm64 -Xcc -I/Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/DerivedSources/arm64 -Xcc -I/Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/DerivedSources -Xcc -DPOD_CONFIGURATION_RELEASE=1 -Xcc -DCOCOAPODS=1 -emit-objc-header -emit-objc-header-path /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/Objects-normal/arm64/SwiftPhoenixClient-Swift.h -import-underlying-module -Xcc -ivfsoverlay -Xcc /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/unextended-module-overlay.yaml -working-directory /Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Example/Pods
error: Build input files cannot be found: '/Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Sources/client/Message.swift', '/Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Sources/client/Utilities/Defaults.swift', '/Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Sources/client/Utilities/Delegated.swift', '/Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Sources/client/TimeoutTimer.swift', '/Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Sources/client/Presence.swift', '/Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Sources/client/Push.swift', '/Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Sources/client/Channel.swift', '/Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Sources/client/Socket.swift' (in target 'SwiftPhoenixClient' from project 'Pods')

CompileSwiftSources normal armv7 com.apple.xcode.tools.swift.compiler (in target 'SwiftPhoenixClient' from project 'Pods')
    cd /Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Example/Pods
    export DEVELOPER_DIR=/Applications/Xcode.app/Contents/Developer
    export SDKROOT=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.2.sdk
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc -incremental -module-name SwiftPhoenixClient -O -whole-module-optimization -enforce-exclusivity=checked @/Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/Objects-normal/armv7/SwiftPhoenixClient.SwiftFileList -D COCOAPODS -suppress-warnings -sdk /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.2.sdk -target armv7-apple-ios9.0 -g -module-cache-path /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/ModuleCache.noindex -Xfrontend -serialize-debugging-options -embed-bitcode -swift-version 5 -I /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/BuildProductsPath/Release-iphoneos/SwiftPhoenixClient -F /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/BuildProductsPath/Release-iphoneos/SwiftPhoenixClient -F /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/BuildProductsPath/Release-iphoneos/Starscream -c -num-threads 8 -output-file-map /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/Objects-normal/armv7/SwiftPhoenixClient-OutputFileMap.json -parseable-output -serialize-diagnostics -emit-dependencies -emit-module -emit-module-path /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/Objects-normal/armv7/SwiftPhoenixClient.swiftmodule -Xcc -I/Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/swift-overrides.hmap -Xcc -iquote -Xcc /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/SwiftPhoenixClient-generated-files.hmap -Xcc -I/Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/SwiftPhoenixClient-own-target-headers.hmap -Xcc -I/Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/SwiftPhoenixClient-all-non-framework-target-headers.hmap -Xcc -ivfsoverlay -Xcc /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/all-product-headers.yaml -Xcc -iquote -Xcc /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/SwiftPhoenixClient-project-headers.hmap -Xcc -I/Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/BuildProductsPath/Release-iphoneos/SwiftPhoenixClient/include -Xcc -I/Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/DerivedSources-normal/armv7 -Xcc -I/Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/DerivedSources/armv7 -Xcc -I/Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/DerivedSources -Xcc -DPOD_CONFIGURATION_RELEASE=1 -Xcc -DCOCOAPODS=1 -emit-objc-header -emit-objc-header-path /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/Objects-normal/armv7/SwiftPhoenixClient-Swift.h -import-underlying-module -Xcc -ivfsoverlay -Xcc /Users/chamithawijesekera/Library/Caches/org.carthage.CarthageKit/DerivedData/11.3_11C29/SwiftPhoenixClient/1.2.0/Build/Intermediates.noindex/ArchiveIntermediates/SwiftPhoenixClient/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/SwiftPhoenixClient.build/unextended-module-overlay.yaml -working-directory /Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Example/Pods
error: Build input files cannot be found: '/Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Sources/client/Message.swift', '/Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Sources/client/Utilities/Defaults.swift', '/Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Sources/client/Utilities/Delegated.swift', '/Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Sources/client/TimeoutTimer.swift', '/Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Sources/client/Presence.swift', '/Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Sources/client/Push.swift', '/Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Sources/client/Channel.swift', '/Users/chamithawijesekera/teamapp/team-app-ios/Carthage/Checkouts/SwiftPhoenixClient/Sources/client/Socket.swift' (in target 'SwiftPhoenixClient' from project 'Pods')

** ARCHIVE FAILED **
```

Any thoughts? Thanks!

**Comment by chamitha** [comment]
Anyone know the cause of this error?

**Comment by dsrees** [comment]
@chamitha sorry for not replying sooner. I am not very familiar with carthage so i'm not much help here at the moment. I'll take some time to look into this today though and see if I can reproduce

**Comment by chamitha** [comment]
Thanks @dsrees

**Comment by chamitha** [comment]
@dsrees Did you get a chance to look into this? Cheers

**Comment by dsrees** [comment]
@chamitha I'm able to reproduce. I'll keep investigating

**Comment by dsrees** [comment]
@chamitha this was caused by Starscream introducing a new API in v4.x that broke the client. I'm releasing a version that pins starscream to the previous 3.x API.

**Comment by chamitha** [comment]
No worries. Thanks for fixing.

**Comment by dsrees** [comment]
Resolved in 1.2.1
